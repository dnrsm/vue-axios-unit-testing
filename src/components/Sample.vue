<template>
  <div class="hello">
    <ul>
      <li>{{ userId }}</li>
      <li>{{ title }}</li>
    </ul>
  </div>
</template>

<script>
export default {
  name: "HelloWorld",
  data() {
    return {
      userId: "",
      title: ""
    };
  },
  created() {
    this.request();
  },
  methods: {
    request() {
      this.$axios
        .get("https://jsonplaceholder.typicode.com/posts/1")
        .then(res => {
          this.userId = res.data.userId;
          this.title = res.data.title;
        });
    }
  }
};
</script>
