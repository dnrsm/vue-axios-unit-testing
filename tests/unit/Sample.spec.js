import { shallowMount } from "@vue/test-utils";
import Sample from "@/components/Sample.vue";
import mockAxios from "axios";

const response = {
  data: {
    userId: 1,
    id: 1,
    title:
      "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
    body:
      "quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto"
  }
};

describe("Sample.vue", () => {
  mockAxios.get.mockReturnValue(Promise.resolve(response));
  const wrapper = shallowMount(Sample, {
    mocks: {
      $axios: mockAxios
    }
  });
  it("userId â‡› 1", () => {
    expect(wrapper.exists()).toBe(true);
    expect(mockAxios.get).toHaveBeenCalledTimes(1);
    expect(mockAxios.get).toBeCalledWith(
      "https://jsonplaceholder.typicode.com/posts/1"
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.vm.userId).toBe(1);
  });
});
