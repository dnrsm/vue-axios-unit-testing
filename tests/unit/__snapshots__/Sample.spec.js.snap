// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sample.vue users[0].name should be Graham 1`] = `
<div class="hello">
  <ul>
    <li>1</li>
    <li>sunt aut facere repellat provident occaecati excepturi optio reprehenderit</li>
  </ul>
</div>
`;
